code/go/0chain.net/smartcontract/dbs/event/process.go

```diff
@@ -86,16 +95,24 @@ func (edb *EventDb) ProcessEvents(
 				zap.String("block", block),
 				zap.Int("block size", blockSize))
 		}
+
+		if !commit {
+			err := tx.Rollback()
+			if err != nil {
+				return nil, err
+			}
```

Comment:
We need to return `nil, error` here. If `commit` is false, that means the event was failed to be processed, and we need to avoid this `tx` to be used later after rollback.